pkg/scheduler/scheduler.go

```diff
@@ -180,15 +180,15 @@ func (cu *cohortsUsage) hasCommonFlavorResources(cohort string, assignment cache
 	return false
 }
 
-func (s *Scheduler) schedule(ctx context.Context) {
+func (s *Scheduler) schedule(ctx context.Context) wait.SpeedSignal {
 	log := ctrl.LoggerFrom(ctx)
 
 	// 1. Get the heads from the queues, including their desired clusterQueue.
 	// This operation blocks while the queues are empty.
 	headWorkloads := s.queues.Heads(ctx)
 	// If there are no elements, it means that the program is finishing.
 	if len(headWorkloads) == 0 {
-		return
+		return wait.SlowDown{}
```

Comment:
`KeepGoing`, because the program is finishing....

If there are no elements in the queues, `s.queues.Heads` simply blocks.